rm-portal/src/test/java/org/ow2/proactive_grid_cloud_portal/rm/client/monitoring/views/compact/CompactFlowPanelTest.java

```diff
@@ -0,0 +1,335 @@
+    }
+
+    @Test
+    public void testAddDeployingNodes() {
+        for (Tile nodeSourceTile : nodeSources(1)) {
+            spyCompactFlowPanel.drawNodeSource(nodeSourceTile);
+            for (Tile nodeTile : deployingNodes(nodeSourceTile.getNodesource().getSourceName(), 10)) {
+                spyCompactFlowPanel.drawNode(nodeTile, null);
+            }
+        }
+
+        assertEquals(11, spyCompactFlowPanel.getTilesNumber());
+    }
+
+    @Test
+    public void testIndexOf() {
+
+        for (Tile nodeSourceTile : nodeSources(1)) {
+            spyCompactFlowPanel.drawNodeSource(nodeSourceTile);
+
+            Tile hostTile = hosts(nodeSourceTile.getNodesource().getSourceName(), 1).get(0);
+            for (Tile nodeTile : nodes(nodeSourceTile.getNodesource().getSourceName(),
+                                       hostTile.getHost().getHostName(),
+                                       20000)) {
+                spyCompactFlowPanel.drawNode(nodeTile, hostTile);
+            }
+
+        }
+
+        final NodeSource.Host.Node node = spyCompactFlowPanel.getModel().get(0).getHosts().get(0).getNodes().get(10000);
+        assertEquals(10002, spyCompactFlowPanel.indexOf(node).get().longValue());
+
+    }
+
+    @Test
+    public void testAddingOneMoreNode() {
+
+        for (Tile nodeSourceTile : nodeSources(5)) {
+            spyCompactFlowPanel.drawNodeSource(nodeSourceTile);
+
+            for (Tile deployingTile : deployingNodes(nodeSourceTile.getNodesource().getSourceName(), 5)) {
+                spyCompactFlowPanel.drawNode(deployingTile, null);
+            }
+
+            for (Tile hostTile : hosts(nodeSourceTile.getNodesource().getSourceName(), 5)) {
+
+                for (Tile nodeTile : nodes(nodeSourceTile.getNodesource().getSourceName(),
+                                           hostTile.getHost().getHostName(),
+                                           5)) {
+                    spyCompactFlowPanel.drawNode(nodeTile, hostTile);
+                }
+            }
+        }
+
+        long totalCount = 5 * (1 + 5 + 5 * (1 + 5));
+
+        assertEquals(totalCount, spyCompactFlowPanel.getTilesNumber());
+
+        final HierarchyNodeSource hierarchyNodeSource = spyCompactFlowPanel.getModel().get(2);
+        final HierarchyHost hierarchyHost = hierarchyNodeSource.getHosts().get(2);
+        final Tile nodeTile = node(hierarchyNodeSource.getNodeSource().getSourceName(),
+                                   hierarchyHost.getHost().getHostName(),
+                                   "newNode");
+        final Tile hostTile = host(hierarchyNodeSource.getNodeSource().getSourceName(),
+                                   hierarchyHost.getHost().getHostName());
+
+        spyCompactFlowPanel.drawNode(nodeTile, hostTile);
+
+        ArgumentCaptor<Tile> tileCaptor = ArgumentCaptor.forClass(Tile.class);
+        //        verify(spyCompactFlowPanel, times(1)).insert(tileCaptor.capture(), eq(12));
```

Comment:
Is this comment needed?

Reply:
No, you are right, I will remove it, thanks